bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/XmlThingTypeProvider.java

```diff
@@ -44,239 +41,44 @@
  * @author Dennis Nobel - Added locale support, Added cache for localized thing types
  * @author Ivan Iliev - Added support for system wide channel types
  * @author Kai Kreuzer - fixed concurrency issues
+ * @author Simon Kaufmann - factored out common aspects into {@link AbstractXmlBasedProvider}
  */
 @Component(immediate = true, property = { "esh.scope=core.xml" })
-public class XmlThingTypeProvider implements ThingTypeProvider {
-
-    private class LocalizedThingTypeKey {
-        public ThingTypeUID uid;
-        public String locale;
-
-        public LocalizedThingTypeKey(ThingTypeUID uid, String locale) {
-            this.uid = uid;
-            this.locale = locale;
-        }
-
-        @Override
-        public int hashCode() {
-            final int prime = 31;
-            int result = 1;
-            result = prime * result + getOuterType().hashCode();
-            result = prime * result + ((locale == null) ? 0 : locale.hashCode());
-            result = prime * result + ((uid == null) ? 0 : uid.hashCode());
-            return result;
-        }
-
-        @Override
-        public boolean equals(Object obj) {
-            if (this == obj) {
-                return true;
-            }
-            if (obj == null) {
-                return false;
-            }
-            if (getClass() != obj.getClass()) {
-                return false;
-            }
-            LocalizedThingTypeKey other = (LocalizedThingTypeKey) obj;
-            if (!getOuterType().equals(other.getOuterType())) {
-                return false;
-            }
-            if (locale == null) {
-                if (other.locale != null) {
-                    return false;
-                }
-            } else if (!locale.equals(other.locale)) {
-                return false;
-            }
-            if (uid == null) {
-                if (other.uid != null) {
-                    return false;
-                }
-            } else if (!uid.equals(other.uid)) {
-                return false;
-            }
-            return true;
-        }
-
-        private XmlThingTypeProvider getOuterType() {
-            return XmlThingTypeProvider.this;
-        }
-
-    }
+public class XmlThingTypeProvider extends AbstractXmlBasedProvider<ThingTypeUID, ThingType>
```

Comment:
There is now
* ConfigXmlConfigDescriptionProvider | esh.scope=core.xml.config
* XmlChannelGroupTypeProvider | esh.scope=core.xml.channelGroups
* XmlChannelTypeProvider | esh.scope=core.xml.channels
* XmlThingTypeProvider | esh.scope=core.xml

To be consistent I would expect the esh.scope should be equal to core.xml.things for the XmlThingTypeProvider (to fir to channel type and channel group type provider).

Why do we hide the "type" at all?